### PR TITLE
Feature - Base - Config base project

### PR DESCRIPTION
**Pull request by at-thinhnguyen-2**
- Add file 
- Config gem

### CODE REVIEW

File: WeatherDay/Library/Ext/DataExt.swift [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+//  DataExt.swift
+//  FinalProject
+//
+//  Created by Thinh Nguyen X. on 9/10/20.
+//  Copyright © 2020 Thinh Nguyen X. All rights reserved.
+//
+
+import Foundation
+extension Data {
+
```
Comment (at-thinhnguyen-2) [inline WeatherDay/Library/Ext/DataExt.swift:11]: WARNING - Lines should not have trailing whitespace.

File: WeatherDay/Library/Ext/StringExt.swift [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        switch method {
+        case .encode:
+            guard let data = data(using: .utf8) else { return nil }
+            return data.base64EncodedString()
+        case .decode:
+            guard let data = Data(base64Encoded: self) else { return nil }
+            return String(data: data, encoding: .utf8)
+        }
+    }
+
```
Comment (at-thinhnguyen-2) [inline WeatherDay/Library/Ext/StringExt.swift:37]: WARNING - Lines should not have trailing whitespace.

File: WeatherDay/Library/Ext/StringExt.swift [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            guard let data = Data(base64Encoded: self) else { return nil }
+            return String(data: data, encoding: .utf8)
+        }
+    }
+    
+    /// Initializes an NSURL object with a provided URL string. (read-only)
+    public var url: URL? {
+        return URL(string: self)
+    }
+
```
Comment (at-thinhnguyen-2) [inline WeatherDay/Library/Ext/StringExt.swift:42]: WARNING - Lines should not have trailing whitespace.

File: WeatherDay/View/AppDelegate.swift [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+//  Copyright © 2020 Thinh Nguyen X. All rights reserved.
+//
+
+import UIKit
+
+@UIApplicationMain
+class AppDelegate: UIResponder, UIApplicationDelegate {
+
+
+
```
Comment (at-thinhnguyen-2) [inline WeatherDay/View/AppDelegate.swift:15]: WARNING - Limit vertical whitespace to a single empty line. Currently 3.

File: WeatherDay/View/AppDelegate.swift [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        return UISceneConfiguration(name: "Default Configuration", sessionRole: connectingSceneSession.role)
+    }
+
+    func application(_ application: UIApplication, didDiscardSceneSessions sceneSessions: Set<UISceneSession>) {
+        // Called when the user discards a scene session.
+        // If any sessions were discarded while the application was not running, this will be called shortly after application:didFinishLaunchingWithOptions.
+        // Use this method to release any resources that were specific to the discarded scenes, as they will not return.
+    }
+
+
```
Comment (at-thinhnguyen-2) [inline WeatherDay/View/AppDelegate.swift:35]: WARNING - Limit vertical whitespace to a single empty line. Currently 2.

File: WeatherDay/View/AppDelegate.swift [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+}
+
```
Comment (at-thinhnguyen-2) [inline WeatherDay/View/AppDelegate.swift:37]: WARNING - Files should have a single trailing newline.

**Comment by at-thinhnguyen-2** [comment]
Total linter violations in pull request: 6
Serious: 0
Warnings: 6

**Comment by at-thinhnguyen-2** [comment]
Total linter violations in pull request: 2
Serious: 0
Warnings: 2

**Comment by at-thinhnguyen-2** [comment]
Total linter violations in pull request: 1
Serious: 0
Warnings: 1

**Comment by at-thinhnguyen-2** [comment]
:+1: There are no linter violations.

**Review by at-khoavo — APPROVED** [review]
LGTM!

**Pull request merged**